Update to the latest semver

jtgeibel opened the pull request:
The new release of semver treats `>= 0` the same as the `*` wildcard.
It makes sense that these are equivalent, but this did break 3 tests
and will technically reject syntax in `Cargo.toml` that was previously
accepted.

r? @ghost

jtgeibel commented on src/tests/builders/dependency.rs:
```diff
@@ -15,7 +15,7 @@ impl DependencyBuilder {
             name: name.to_string(),
             registry: None,
-            version_req: u::EncodableCrateVersionReq(semver::VersionReq::parse(">= 0").unwrap()),
+            version_req: u::EncodableCrateVersionReq(semver::VersionReq::parse("> 0").unwrap()),
```
Added to the weekly meeting to discuss if disallowing `>= 0` (as we do for the equivalent `*` wildcard) is an acceptable change.

bors commented:
:umbrella: The latest upstream changes (presumably #3023) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#conflicts).

*Note that reviewers usually do not review pull requests until merge conflicts are resolved!* Once you resolve the conflicts, you should change the labels applied by bors to indicate that your PR is ready for review. Post this as a comment to change the labels:
```
@rustbot modify labels: +S-waiting-on-review -S-waiting-on-author
```

jtgeibel commented:
I've rebased and updated the error message. I've also run the following commands to determine crates already using a `>= 0` constraint.

```console
$ cd ~/.cargo/registry/index/github.com-1ecc6299db9ec823/
$ git grep --count '">= 0"' origin/master
origin/master:as/ki/askii:1
origin/master:co/ar/coarsetime:7
origin/master:co/un/count-min-sketch:2
origin/master:cu/rl/curl-sys:31
origin/master:cu/rs/cursive_buffered_backend:4
origin/master:do/h-/doh-proxy:8
origin/master:ju/mp/jumphash:4
origin/master:li/bg/libgit2-sys:38
origin/master:li/bs/libssh2-sys:16
origin/master:na/ts/nats:5
origin/master:pr/iv/privdrop:5
origin/master:vm/-m/vm-memory:2
$ git grep --count '">=0"' origin/master
origin/master:js/on/jsonschema:1
origin/master:vm/-m/vm-memory:5
```

I've checked the index for all of these, and `jsonschema` and `vm-memory` are the only 2 crates where the latest release uses the bad constraint. `jsonschema` has a dev-dependency on `mockito >=0` and `vm-memory` a dev-dependency on `matches >=0`.

r? @pietroalbini

pietroalbini commented:
@bors r+

bors commented:
:pushpin: Commit 4819d905ab95b4029f4dc9cc5bd9b04e0e57b524 has been approved by `pietroalbini`

<!-- @bors r=pietroalbini 4819d905ab95b4029f4dc9cc5bd9b04e0e57b524 -->
<!-- homu: {"type":"Approved","sha":"4819d905ab95b4029f4dc9cc5bd9b04e0e57b524","approver":"pietroalbini"} -->

bors commented:
:hourglass: Testing commit 4819d905ab95b4029f4dc9cc5bd9b04e0e57b524 with merge cdf69158ebf5ece20aa9aceca83aefe8991562d4...
<!-- homu: {"type":"BuildStarted","head_sha":"4819d905ab95b4029f4dc9cc5bd9b04e0e57b524","merge_sha":"cdf69158ebf5ece20aa9aceca83aefe8991562d4"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang/crates.io/runs/1430290433)
Approved by: pietroalbini
Pushing cdf69158ebf5ece20aa9aceca83aefe8991562d4 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"pietroalbini","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang/crates.io/runs/1430290433"},"merge_sha":"cdf69158ebf5ece20aa9aceca83aefe8991562d4"} -->

dtolnay commented:
Is it possible that this update regressed the rendering of dependency version requirements? https://github.com/rust-lang/crates.io/issues/3047

Turbo87 commented:
that does seem likely, yeah. certainly not an intentional change. 😞
